lib/ecom/api.ts

```diff
@@ -285,8 +278,32 @@ function createEcomApi(wixClient: WixApiClient): EcomApi {
                 return failureResponse(EcomApiErrorCodes.GetProductsFailure, getErrorMessage(e));
             }
         },
-    };
-}
+    });
+
+/**
+ * Wraps all methods of the EcomApi with a try-catch block that fixes broken
+ * error messages in WixClient errors and rethrows them.
+ */
+const withNormalizedWixClientErrors = (api: EcomApi): EcomApi => {
```

Comment:
I'd recomment to move this to a separate file. I understand that for now it is not needed, but I that API growth, so I can image we will decide to split it to few smaller sub-APIs. in that case this will help

Reply:
Since this function mutates the original object, I only feel comfortable defining it next to the object it mutates. Using this function as a black box without understanding its implementation could lead to tricky bugs.

For example, if EcomApi was defined as a class:
```ts
class EcomApi {
    getCart() { /* ... */ }
}
```

TypeScript would happily let you do:
```ts
const api = withNormalizedWixClientErrors(new EcomApi());
```

But later you would discover that error normalization has stopped working.

To solve the issues with mutation and non-iterable properties I could use a proxy, but it has more subtle issues, such `ecomApi.getOrder` not being a stable reference and returning a new function each time you access it.

If we're going to split `EcomApi` into multiple files, I think one of the alternative approaches outlined in the PR description would be safer and more predictable.